### DISCUSSION TITLE
Invalidating and Bypassing routeRules cache

### DISCUSSION THREAD

**Discussion by dolbex** [nuxt/nuxt#20495] [Questions]
Hey folks. I've been looking for a solution to this for the last couple of days and I'm a little surprised i'm not seeing more folks seeking this out and there doesn't currently seem to be an answer. 

**Question**: How do I invalidate or bypass a full page using routeRules? Or, is there another way to achieve page cache that is bypassable and can be invalidated on demand?

This is totally possible in the server/routes via something like:

```js
export default cachedEventHandler((event) => {
  return `Response generated at ${new Date().toISOString()}`
}, { swr: false, maxAge: 10, shouldBypassCache: () => true })
```

But if I do something like this which will cache all page routes for an hour there is no way to clear or bypass the cache:

```js
  nitro: {
    // preset: 'aws-lambda',
    routeRules: {
      '/**': {
        cache: {
          swr: true,
          maxAge: 60 * 60,
        },
      },
    },
  },
```

I totally recognize that *"This feature is still experimental and may evolve in the future."* - I just want to confirm I'm not missing anything and that my statements above are true. As a bonus: can someone confirm feature is coming down the pipe.

**Comment by maou-shonen** [comment]
https://nitro.unjs.io/guide/cache#route-rules
 I guess that your issue could be resolved by changing the cache storage layer to another driver that you can manipulate(such as Redis),
and then clearing the cache when you need to?

**Reply by dolbex** [reply]
@mao-shonen - definitely considered that but it doesn't allow me to conditionally bypass the cache. I think, if my sus above are confirmed, I will utilize cloudflare caching with workers to do it.

**Reply by maou-shonen** [reply]
Then you may need to implement your own cache page...
Fortunately, a combination of custom routeRules, middleware, and the render:response hook should make it easy to do?

**Reply by maou-shonen** [reply]
I also saw this [todo](https://github.com/unjs/nitro/blob/7a6c5b46da627acb1a565211c712d37794df5adb/src/runtime/renderer.ts#L44).

**Reply by dolbex** [reply]
> Then you may need to implement your own cache page... Fortunately, a combination of custom routeRules, middleware, and the render:response hook should make it easy to do?

Good find on the Todo

I'm actually not sure I understand how I would do that with the tools you're suggesting. I don't have any access to the routeRules cache in middleware. Do you have any more insight there? Also, thanks for chiming in - appreciate the attention!

**Reply by maou-shonen** [reply]
`routeRules` can actually store any parameters and you can retrieve them in Nitro by using `getRouteRules`.
